src/StudioHeader.test.jsx

```diff
@@ -0,0 +1,233 @@
+                LOGO_URL: process.env.LOGO_URL,
+              },
+            }}
+          >
+            <StudioHeader />
+          </AppContext.Provider>
+        </IntlProvider>
+      </ResponsiveContext.Provider>
+    );
+
+    const wrapper = TestRenderer.create(component);
+
+    expect(wrapper.toJSON()).toMatchSnapshot();
+  });
+
+  it('renders correctly for authenticated desktop', () => {
+    const component = (
+      <ResponsiveContext.Provider value={{ width: 1280 }}>
+        <IntlProvider locale="en" messages={{}}>
+          <AppContext.Provider
+            value={{
+              authenticatedUser: {
+                userId: 'abc123',
+                username: 'edX',
+                roles: [],
+                administrator: false,
+              },
+              config: {
+                STUDIO_BASE_URL: process.env.STUDIO_BASE_URL,
+                SITE_NAME: process.env.SITE_NAME,
+                LOGIN_URL: process.env.LOGIN_URL,
+                LOGOUT_URL: process.env.LOGOUT_URL,
+                LOGO_URL: process.env.LOGO_URL,
+              },
+            }}
+          >
+            <StudioHeader />
+          </AppContext.Provider>
+        </IntlProvider>
+      </ResponsiveContext.Provider>
+    );
+
+    const wrapper = TestRenderer.create(component);
+
+    expect(wrapper.toJSON()).toMatchSnapshot();
+  });
+
+  it('renders correctly for an anonymous desktop, with an app menu', () => {
+    const appMenu = {
+      content: 'App Menu',
+      href: 'https://href-url.org',
+      menuItems: [
+        {
+          menuType: 'dropdown',
+          menuHref: 'https://menu-href-url.org',
+          menuContent: 'Content 1',
+        },
+        {
+          menuType: 'dropdown',
+          menuHref: 'https://menu-href-url.org',
+          menuContent: 'Content 2',
+        },
+        {
+          menuType: 'dropdown',
+          menuHref: 'https://menu-href-url.org',
+          menuContent: 'Content 3',
+        },
+      ],
+    };
+    const component = (
+      <ResponsiveContext.Provider value={{ width: 1280 }}>
+        <IntlProvider locale="en" messages={{}}>
+          <AppContext.Provider
+            value={{
+              authenticatedUser: {
+                userId: 'abc123',
+                username: 'edX',
+                roles: [],
+                administrator: false,
+              },
+              config: {
+                STUDIO_BASE_URL: process.env.STUDIO_BASE_URL,
+                SITE_NAME: process.env.SITE_NAME,
+                LOGIN_URL: process.env.LOGIN_URL,
+                LOGOUT_URL: process.env.LOGOUT_URL,
+                LOGO_URL: process.env.LOGO_URL,
+              },
+            }}
+          >
+            <StudioHeader appMenu={appMenu} />
+          </AppContext.Provider>
+        </IntlProvider>
+      </ResponsiveContext.Provider>
+    );
+
+    const wrapper = TestRenderer.create(component);
+
+    expect(wrapper.toJSON()).toMatchSnapshot();
+  });
```

Comment:
As noted above, don't think there's a need to test anonymous access, here.  It's not likely to ever happen.

Reply:
tests removed